container-disc/src/main/java/com/yahoo/container/jdisc/metric/JrtMetrics.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.container.jdisc.metric;
+
+import com.yahoo.jdisc.Metric;
+import com.yahoo.jrt.TransportMetrics;
+
+import java.util.function.ToLongFunction;
+
+import static com.yahoo.jrt.TransportMetrics.*;
+
+/**
+ * Emits jrt metrics
+ *
+ * @author bjorncs
+ */
+class JrtMetrics {
+
+    private final TransportMetrics transportMetrics = TransportMetrics.getInstance();
+    private final Metric metric;
+    private volatile Snapshot previousSnapshot;
```

Comment:
either this should not need to be volatile or we need more synchronization elsewhere (the pattern is expected to be local cache for single writer)